app/src/main/java/com/zulip/android/activities/ZulipActivity.java

```diff
@@ -2043,6 +2043,8 @@ public void run() {
      */
     private void openLogin() {
         Intent i = new Intent(this, LoginActivity.class);
+        String server_url = getSharedPreferences(LoginActivity.Prefs_name,MODE_PRIVATE).getString(LoginActivity.Prefs_server,null);
```

Comment:
You will have to pull out the serverUrl before app.logOut(); as it will flush out the value 
Also use ZulipApp.get().getServerURI()